libV2/tasks/DiskUsage.js

```diff
@@ -166,6 +206,18 @@ class MonitorDiskUsage extends BaseTask {
                 await this._expireMetrics();
             }
         }
+
+        if (this._hardLimit !== undefined) {
+            const shouldLock = this._checkHardLimit(size, this.nodeId);
+            if (shouldLock) {
+                moduleLogger.error('hard limit exceeded, disabling writes to warp 10', { nodeId: this.nodeId });
+                await this._disableWarp10Updates();
+            } else {
+                moduleLogger.error('usage below hard limit, ensuring writes to warp 10 are enabled',
+                    { nodeId: this.nodeId });
```

Comment:
At some point we said that we should not re-enable writes automatically (since we lost events, and repair is needed, etc.)

Reply:
Ah I forgot about that. #899 removes the technical barriers to automatic unlocking as far as lost events/consistency, though I think there may have been an ops reason as well? Either way I'll add an manual unlock flag to the cli

Reply:
if you say so

Reply:
No risk of inconsistency if let say, 2 nodes were locked for 2 days ?

Reply:
Under the new system any events that were destined to be written to the locked warp 10s would have been written to one of the available nodes and processed there. The snapshot/checkpoint/repair processes already allow for arbitrary lengths of time in between them so from their perspective a locked warp 10 would look the same as an idle cluster.

Reply:
Not counting the lag in replication between warp 10s it would introduce of course